### PR TITLE
Refactor color tokens for theming capabilities

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #296 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

* Update dependencies
* Refactor color tokens by isolating the color tokens and creating tier 3 tokens for repeatable rules
* Replace hardcoded size values with tokens

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

## Verification
- [x] No linter issues
- [x] Build:release completes without error
- [x] New tier 3 component tokens exist
- [x] API Page "Theme" section is valid

UI
- [x] All input texts are correct color in all states
- [x] All help texts are correct color in all states
- [x] All label texts are correct color in all states
- [x] Caret color is correct shade of blue
- [x] Type Icon color is correct color in all states
- [x] Alert icon is correct color
- [x] Clear icon is correct color
- [x] All border colors are correct color in all states
- [x] All background colors are correct color

Figma: https://www.figma.com/design/tSkHw1uJfDHy0Ml0aDp7Io/branch/SWOHKZpz2K78PPL43uvp5z/Auro-color-palette-(Variables)?node-id=4224-10856&m=dev

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: src/tokens.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+:host {
+  --ds-auro-input-background-color: var(--ds-color-background-primary-100-default, $ds-color-background-primary-100-default);
+  --ds-auro-input-border-color: var(--ds-color-border-primary-default, $ds-color-border-primary-default);
+  --ds-auro-input-text-color: var(--ds-color-text-secondary-default, $ds-color-text-secondary-default);
+  --ds-auro-input-alert-icon-color: var(--ds-color-alert-error-default, $ds-color-alert-error-default);
+  --ds-auro-input-type-icon-color: var(--ds-color-text-disabled-default, $ds-color-text-disabled-default);
```
Comment (jason-capsule42) [inline src/tokens.scss:6]: I see input text color (I assume thats for the value in the actual HTML5 input) but we also need to be able to style the label text color and the help text color.

It looks like you are using the same token value for all of them but it should be 3 different ones.

File: src/color.scss [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+  &::placeholder {
+    color: transparent;
+  }
+
+  &:focus {
+    &::placeholder {
+      color: var(--ds-auro-input-text-color);
+    }
+  }
```
Comment (jason-capsule42) [inline src/color.scss:37]: placeholder text should probably have it's own tier 3 token which defaults to transparent and then change the value during focus.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

File: src/tokens.scss [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  --ds-auro-input-background-color: var(--ds-color-background-primary-100-default, $ds-color-background-primary-100-default);
+  --ds-auro-input-border-color: var(--ds-color-border-secondary-default, $ds-color-border-secondary-default);
+  --ds-auro-input-caret-color: var(--ds-color-text-ui-focus-default, $ds-color-text-ui-focus-default);
+  --ds-auro-input-alert-icon-color: var(--ds-color-alert-error-default, $ds-color-alert-error-default);
+  --ds-auro-input-notification-icon-color: var(--ds-color-alert-error-default, $ds-color-alert-error-default);
+  --ds-auro-input-type-icon-color: var(--ds-color-icon-primary-default, $ds-color-icon-primary-default);
+  --ds-auro-input-text-color: var(--ds-color-text-primary-default, $ds-color-text-primary-default);
+  --ds-auro-input-placeholder-text-color: transparent;
+  --ds-auro-input-label-text-color: var(--ds-color-text-tertiary-default, $ds-color-text-tertiary-default);
+  --ds-auro-input-help-text-color: var(--ds-color-text-tertiary-default, $ds-color-text-tertiary-default);
```
Comment (jason-capsule42) [inline src/tokens.scss:11]: Let's keep this list alphabetical to make it a little easier to read.

File: package.json [hunk 4]
```diff
@@ -149,7 +149,7 @@
     "build:dev:assets": "npm-run-all build:sass:component postCss:component sass:render",
     "build:docs": "node scripts/generateDocs.mjs",
     "build:sass": "npm-run-all build:sass:component postCss:component sass:render",
-    "build:sass:component": "sass --no-source-map src/style.scss:src/style.css",
+    "build:sass:component": "for file in src/*.scss; do npx sass --no-source-map \"$file:${file%.scss}.css\"; done",
```
Comment (jason-capsule42) [inline package.json:152]: This is interesting and clever, make sure to open a ticket in the generator to migrate this there! Nice work.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (blackfalcon) [reply package.json:152]: I saw the Discussion on updating this. I'd like to understand this more on why we now need a `for` loop and the use of `npx` over a local dependency? 

https://github.com/orgs/AlaskaAirlines/discussions/557#discussioncomment-9799161

File: src/color.scss [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+// Handle highlighting the border during focus and invalid state
+@mixin border-highlight {
+  &:before {
+    border-bottom-color: transparent;
+  }
+}
+
+.wrapper {
+  border-color: transparent;
```
Comment (blackfalcon) [inline src/color.scss:18]: So the idea here is to move all the color CSS into it's one stylesheet? Won't that make it difficult to maintain? 

What is the value of moving this to its own stylesheet versus having these styles in line with the others?

File: package.json (hunk 4 above)
Comment (jordanjones243) [reply package.json:152]: Maybe we don't have an inherent need for this in all of our components. This script was my solution to the issue where the `color.scss` and `tokens.scss` file would not compile.

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**